datadog_checks_base/datadog_checks/base/checks/windows/perf_counters/counter.py

```diff
@@ -0,0 +1,543 @@
+            name, config, check, connection, object_name, metric_prefix, counter_selector, tags, custom_transformer
+        )
+
+        self.tag_name = tag_name
+
+        self.average = config.get('average', False)
+        if not isinstance(self.average, bool):
+            raise ConfigTypeError(
+                f'Option `average` for counter `{self.name}` of performance '
+                f'object `{self.object_name}` must be a boolean'
+            )
+
+        self.aggregate_transformer = None
+        self.aggregate = config.get('aggregate', False)
+        if not isinstance(self.aggregate, bool) and self.aggregate != 'only':
+            raise ConfigTypeError(
+                f'Option `aggregate` for counter `{self.name}` of performance '
+                f'object `{self.object_name}` must be a boolean or set to `only`'
+            )
+        elif self.aggregate is not False:
+            if self.metric_type not in NATIVE_TRANSFORMERS:
+                raise ConfigTypeError(
+                    f'Option `aggregate` for counter `{self.name}` of performance object `{self.object_name}` is '
+                    f'enabled so `type` must be set to one of the following: {", ".join(NATIVE_TRANSFORMERS)}'
+                )
+            else:
+                metric_name = f'{self.metric_name}.avg' if self.average else f'{self.metric_name}.sum'
+                self.aggregate_transformer = NATIVE_TRANSFORMERS[self.metric_type](check, metric_name, config)
+
+        # All monitored counter handles keyed by the instance name
+        self.instances = {}
+
+    def collect(self):
+        handles_with_data = 0
+        total = 0
+        for instance, counter_handles in self.instances.items():
+            # Some instances may not have a value yet, so we only
+            # use the ones that do for computing averages
+            instance_handles_with_data = len(counter_handles)
+            instance_total = 0
+
+            for i, counter_handle in enumerate(counter_handles):
+                try:
+                    value = get_counter_value(counter_handle)
+                except pywintypes.error as error:
+                    instance_handles_with_data -= 1
+                    self.handle_counter_value_error(error, format_instance(instance, i))
+                else:
+                    instance_total += value
+
+            if not instance_handles_with_data:
+                continue
+
+            handles_with_data += instance_handles_with_data
+            total += instance_total
+
+            if self.aggregate != 'only':
+                tags = [f'{self.tag_name}:{instance}']
+                tags.extend(self.tags)
+
+                if self.average:
+                    self.transformer(instance_total / instance_handles_with_data, tags=tags)
+                else:
+                    self.transformer(instance_total, tags=tags)
+
+        if not handles_with_data:
+            return
+
+        if self.aggregate is not False:
```

Comment:
```suggestion
        if self.aggregate is True:
```
Small nit

Reply:
this is deliberate b/c it can also be set to `only`